REGRESSION(252541@main): [ macOS ] fast/css/style-element-process-crash.html is a flaky timeout

rniwa opened the pull request:
#### d99cbe6bf313671f0181688fe9f598fa439f1314
<pre>
REGRESSION(252541@main): [ macOS ] fast/css/style-element-process-crash.html is a flaky timeout
<a href="https://bugs.webkit.org/show_bug.cgi?id=244013">https://bugs.webkit.org/show_bug.cgi?id=244013</a>

Reviewed by Alexey Proskuryakov.

Delete the test since it&apos;s no longer relevant after <a href="https://commits.webkit.org/242794@main.">https://commits.webkit.org/242794@main.</a>

* LayoutTests/fast/css/style-element-process-crash-expected.txt: Removed.
* LayoutTests/fast/css/style-element-process-crash.html: Removed.
* LayoutTests/platform/mac/TestExpectations:

Canonical link: <a href="https://commits.webkit.org/253590@main">https://commits.webkit.org/253590@main</a>
</pre>













<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9c766150c5fbb8922b539cd91f0b842a39184f3e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/86463 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/30386 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/17429 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/95308 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/149019 "Built successfully and passed tests") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/28789 "Built successfully") | [✅ ~~🛠 mac-debug~~](https://ews-build.webkit.org/#/builders/71/builds/25364 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/78618 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/90552 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/92079 "Passed tests") | [✅ ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/68/builds/23356 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/61/builds/73430 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/23411 "Passed tests") | 
| | [✅ ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/78364 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/78750 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/66412 "Build is being retried") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/26695 "Built successfully") | [✅ ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/12600 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/26609 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/13612 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/28288 "Built successfully") | [✅ ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/61/builds/73430 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/988 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/28229 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/32895 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on LayoutTests/fast/css/style-element-process-crash.html:
```diff
@@ -11,6 +11,7 @@
     if (window.testRunner)
         testRunner.notifyDone();
 }
+onload = () => setTimeout(crash, 0);
```
This test covers a crash that happened when "StyleSheetContents [was] blown away in script execution inside sheetLoaded()." Does it still catch the original issue when executing from onload? If this is still a good test, then we should remove `onfocus="crash()"` below.

Also, we already have waitUntilDone here, and autofocus/onfocus. Doesn't the timeout mean that the input is not focused at all any more? That's more than a timing change.

rniwa replied:
This test isn't testing what it claims to test at least since https://commits.webkit.org/242794@main.
I'd be in the favor of removing it if anything.

aproskuryakov replied:
I think that's acceptable. Trying to have tests for every RefPtr in the code base is futile.

aproskuryakov reviewed (approved):
Please delete style-element-process-crash-expected.txt, too.

rniwa committed:
REGRESSION(252541@main): [ macOS ] fast/css/style-element-process-crash.html is a flaky timeout
https://bugs.webkit.org/show_bug.cgi?id=244013

Reviewed by Alexey Proskuryakov.

Delete the test since it's no longer relevant after https://commits.webkit.org/242794@main.

* LayoutTests/fast/css/style-element-process-crash-expected.txt: Removed.
* LayoutTests/fast/css/style-element-process-crash.html: Removed.
* LayoutTests/platform/mac/TestExpectations:

Canonical link: https://commits.webkit.org/253590@main
